### ISSUE TITLE
Controller 'mdTable', required by directive 'mdColumn', can't be found!

### ISSUE THREAD

**Issue by egantz** [daniel-nagy/fixed-table-header#12]
tried to insert the `fix-head` directive where the `md-head` is and I get this error: `[$compile:ctreq] Controller 'mdTable', required by directive 'mdColumn', can't be found!`

What am I doing wrong?

Using md-data-table v0.9.14

**Comment by daniel-nagy** [comment]
Have you tried reproducing this in a Codepen?

**Comment by leocaseiro** [comment]
I'm having the same issue.

I realized that because I'm using `ui-router`.

I simulate using ng-if and I could reproduce the same error: http://plnkr.co/edit/gb6MIbwF3lCXhx9GUCvZ

I believe if we fix with `ng-if`, it'll fix for ui router and so on.

> UPDATE: What is interesting is if we remove the `fix-head` directive, the header shows on ng-if without any errors/warnings

**Comment by egantz** [comment]
@leocaseiro Exactly. I am using `ui-router` in my app as well and I guess this is the reason why I couldn't reproduce this issue on codepen before. haven't thought ``ui-router` would cause the clash.

@daniel-nagy any ideas? :)

**Comment by daniel-nagy** [comment]
The `fix-head` directive doesn't work with `ng-if` at the moment. If I can find a way to make it work with `ng-if` it might fix this issue.

**Comment by jraadt** [comment]
Yep I believe this is a duplicate of #7 .  I've tried to fix it to no avail... :(

**Comment by leocaseiro** [comment]
I've tried too, not luck.

**Comment by jraadt** [comment]
I now have it working in my pull request #15 where a table's parent element can have an ng-if.  I reworked fixed-table-header to use Angular Material's $mdSticky service, taking cues from md-subheader.  Hope this helps.

**Comment by lucasmaj** [comment]
Sill experiencing this issue. Was there ever a permanent fix?

**Comment by egantz** [comment]
Still happens, couldn't get to make it work :(

**Comment by panickerr** [comment]
@daniel-nagy Even i tired many fixes that were suggested in your Gist and what i knew but the header failed to remain static. Also when i use fix-head in the md-data-tables then the header name disappears.

**Comment by mikila85** [comment]
same here please fix..

**Comment by mennya** [comment]
It works this `ng-show`

**Comment by leocaseiro** [comment]
Has Anyone tried to implement with https://github.com/soenkekluth/sticky-state ?

**Comment by egantz** [comment]
@leocaseiro Yeah, didn't work for me. If anyone succeeds, please share. 👍🏻

**Comment by jrcollins4** [comment]
Running into this issue as well.  Using this with md-data-table v 0.9.4.

**Comment by oviniciusfeitosa** [comment]
Didn't work for me too...
